src/common/from_json.rs

```diff
@@ -89,6 +89,12 @@ impl<T> FromJson for Box<T> where T: FromJson {
     }
 }
 
+impl FromJson for () {
```

Comment:
What's up with this impl? Where was this needed?

Reply:
It is used for [append](https://github.com/QuietMisdreavus/twitter-rs/pull/26/files#diff-03482700cfc921c8cb80b5d988440483R140) since there is no meaningful response (user is supposed to use status to check current state of upload)

Reply:
In that case, i'd much rather see this impl verify that the response is empty (probably by overriding `from_str` and ensuring `.trim().is_empty()` before returning `Ok(())`, and seeing what happens when you try to instance a `Json` from an empty string and changing the `from_json` impl accordingly) instead of just accepting everything. If the endpoint is supposed to return an empty response, i'd like to ensure that it does.

Reply:
Hm... I'm not sure i'm following you. Though I'm not even sure how we'll parse response with empty body, now I think about it...

Reply:
I imagine it'd be something like:

```rust
impl FromJson for () {
    fn from_json(input: &json::Json) -> Result<Self, error::Error> {
        Err(InvalidResponse("Expected empty response", Some(input.to_string())))
    }

    fn from_str(input: &str) -> Result<Self, error::Error> {
        if input.trim().is_empty() {
            Ok(())
        } else {
            Err(InvalidResponse("Expected empty response", Some(input.to_string())))
        }
    }
}
```

Since the response-parsing goes through `from_str` first, overriding that gives it a chance to bypass the json parsing. I'm not sure how `rustc_serialize` handles parsing an empty string, so if it actually gives a real `Json` instance when it's parsed, it may be useful to adjust `from_json` to handle that.

Reply:
Ok, I'm really not familiar with `rustc_serialize`